### DISCUSSION TITLE
W10 Explorer is experiencing a memory leak, and I'm not sure if Open-Shell is the culprit.

### DISCUSSION THREAD

**Discussion by XanthosGambit** [Open-Shell/Open-Shell-Menu#1219] [Q&A]
If I leave my PC running and not doing anything more demanding than running a screensaver, I can come back 8 hours later and find that _explorer.exe_ is using 12-18 GB of system memory. Frankly that seems ridiculous. I figure I'd ask here since Open-Shell directly modifies the Windows 10 GUI and associated processes, but if Open-Shell isn't the culprit, hopefully someone can help me track it down. And I know it basically means nothing, but I've got another computer running Open-Shell that's never experienced this. Maybe I'm cursed or something. Here's my current W10 info:

- Edition:    Windows 10 Pro
- Version:    22H2
- Installed on:    1/‎28/‎2022
- OS build:    19045.2251
- Experience:    Windows Feature Experience Pack 120.2212.4180.0

**Comment by ge0rdi** [comment]
> I'm not sure if Open-Shell is the culprit

Try to uninstall `Open-Shell`, reboot and check if you can replicate big memory usage.

**Reply by XanthosGambit** [reply]
Looks like Open-Shell was the culprit. After uninstalling it and rebooting, _explorer.exe_ is staying at around 350 MB of memory usage.

Guess I'll have to get used to the native W10 Start Menu.

**Reply by ge0rdi** [reply]
You may try to install `Open-Shell` back (use latest version) to confirm that the leak will occur again.

If you are able to replicate the leak in reasonable way (like right after start, or after you do some action with start menu or so), we may try to gather heap trace to figure out what's leaking actually.

**Reply by XanthosGambit** [reply]
Okay, reinstalled Open-Shell using the latest version, and memory usage is climbing. I'm not sure if it's as fast as before, since I never kept any sort of eye on it, but I installed Open-Shell and rebooted about an hour ago. At that time, my _explorer.exe_ memory usage was around 350 MB. It's now 8:23 AM and it's gone up to 715 MB.

EDIT: It's gone down to 415 MB. Re-installing Open-Shell may have actually fixed whatever the issue was. I'll let the computer run for more that an hour and let you know if anything changes.

**Reply by XanthosGambit** [reply]
It's started doing it again.

**Reply by ge0rdi** [reply]
Do you know what triggered it?
Can you replicate it in _reasonable_ time (I mean like not waiting for 3 days to the issue to occur)?

Also, maybe you can uninstall `Open-Shell` again and then use your machine without it for several days.
Since the leak occurred after several days, it is still not clear whether it is `Open-Shell` related or not.

**Reply by XanthosGambit** [reply]
It actually only takes a couple hours of just using the computer for it to happen, I just haven't bothered keeping a close eye on it. And I don't have a clue what triggers it.

**Reply by ge0rdi** [reply]
You can try to obtain heap ETW trace following these steps (all commands have to be done in administrator console):

- Run `wpr -HeapTracingConfig explorer.exe enable`
- Restart Explorer (or whole machine)
- Wait until the leak starts to manifest
- Run `wpr -start Heap -filemode`
- Use your machine normally for couple of minutes (but not for too long because the log will get big)
- Run `wpr -stop heap.etl`
- This will create ETW log in `heap.etl` file
- Upload the log somewhere (Google drive, etc)
- Send me a private link to it to `ge0rdi (at) gmx.com` (I'd not recommend sharing it publicly as it may contain sensitive information)

I'll try to figure out something from it.

**Reply by XanthosGambit** [reply]
Okay, I sent you an email.
